### PR TITLE
[MAINTENANCE] Update `docs snippets` stage in `ci.yml` for scheduled and triggered runs

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What has changed? 
* `RUN_ON_PR_AND_SCHEDULE_AND_TRIGGER` variable for running tests only on non-Fork PRs, schedule, and triggers

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Shinnnyshinshin) [inline .github/workflows/ci.yml:244]: The same pattern (`${{env.RUN_ON_PR_AND_SCHEDULE_AND_TRIGGER}}`)didn't work here. Would like to merge and unblock first and improve iteratively

File: tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_data_on_gcs_using_spark.py
Comment (Shinnnyshinshin) [inline tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_data_on_gcs_using_spark.py]: This change ensures that the `docs-snippets` stage is run. It will be removed before merging.

File: .github/workflows/ci.yml
Comment (anthonyburdi) [inline .github/workflows/ci.yml]: Maybe we can consider changing the name of this variable? If it is not running on fork PRs then we can remove the word FORK.

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml]: shoot you're right.

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml]: PR has been updated

**Review by anthonyburdi — APPROVED** [review]

**Pull request merged**
